Bluepill-runner/Bluepill-runner/BPReportCollector.m

```diff
@@ -51,21 +60,32 @@ + (void)collectReportsFromPath:(NSString *)reportsPath
                     // Test results from other BP workers should continue to parse
                     continue;
                 }
-
-                // Don't withhold the parent object.
-                @autoreleasepool {
-                    NSArray *testsuitesNodes =  [doc nodesForXPath:[NSString stringWithFormat:@".//%@", @"testsuites"] error:&error];
-                    for (NSXMLElement *element in testsuitesNodes) {
+                
+                NSArray *testsuiteNodes = nil;
+                if (XQuery) {
+                    testsuiteNodes = [doc objectsForXQuery:[NSString stringWithFormat:@".//testsuites/testsuite/testsuite/%@/../..", XQuery] error:&error];
```

Comment:
Can you add a comment explaining the XQuery syntax here? I am vaguely familiar with the XPath selectors but I don't understand why three `/testduites` are needed. Thanks!

Reply:
The way Bluepill emits the xml from the JUnit parser creates a <testsuites> for each bundle, and <testsuite> for each test file, and then a <testsuite> for each function inside the test file. But yes I'll add a comment here to be clear.